src/ORM/DataObject.php

```diff
@@ -139,6 +139,9 @@ class DataObject extends ViewableData implements DataObjectInterface, i18nEntity
      */
     private static $default_classname = null;
 
+    private static $reserved_field_names = [
```

Comment:
I prefer the method_exists to reduce the risk of false-negatives. Maybe we should be targeting 5.x in any case?

Reply:
I agree actually, we can method exists check and have a warning/exception when one is found. In terms of target branches, it's a tricky one because using DB fields with a name that already exists as a public method won't work when you use it anyway, but it may be safer to target 5.x anyway